src/com/facebook/buck/jvm/java/intellij/IjProjectWriter.java

```diff
@@ -109,6 +110,15 @@ private Path writeModule(IjModule module) throws IOException {
         "androidFacet",
         projectDataPreparer.getAndroidProperties(module));
 
+    Optional<String> javaSdk = buckConfig.getValue("intellij", "java_sdk");
```

Comment:
IjProjectWriter should cary as little logic and behavior as possible as it's only covered by integration tests. Move this to IjProjectTemplateDataPreparer and add tests.

Also, passing in the entire BuckConfig is overkill as, again, it makes things harder to test (I know the IJProjectCleaner does it, however, you will notice that that class has no tests :/). We have two patterns:
 1) create a IjProjectBuckConfig and have it wrap BuckConfig (see `ArtifactCacheBuckConfig` for an example), this has the benefit of keeping the config-ish logic in one place, but to test you still need to new up a BuckConfig,
 2) create a data object for the config and have separate code that extracts it from BuckConfig (see `AbstractRageConfig`, and `RageBuckConfig`).
I'd prefer using 2), as it makes testing simpler, but if you go 1) I won't block the review.